api/src/main/java/io/druid/data/input/impl/PrefetchableTextFilesFirehoseFactory.java

```diff
@@ -81,26 +82,26 @@
  * disabled, the behavior of the firehose is different like below.
  *
  * <ol>
- *   <li>
- *     If prefetch is enabled, PrefetchableTextFilesFirehose can fetch input objects in background.
- *   </li>
- *   <li> When next() is called, it first checks that there are already fetched files in local storage.
- *     <ol>
- *       <li>
- *         If exists, it simply chooses a fetched file and returns a {@link LineIterator} reading that file.
- *       </li>
- *       <li>
- *         If there is no fetched files in local storage but some objects are still remained to be read, the firehose
- *         fetches one of input objects in background immediately. If an IOException occurs while downloading the object,
- *         it retries up to the maximum retry count. Finally, the firehose returns a {@link LineIterator} only when the
- *         download operation is successfully finished.
- *         </li>
- *     </ol>
- *   </li>
- *   <li>
- *     If prefetch is disabled, the firehose returns a {@link LineIterator} which directly reads the stream opened by
- *     {@link #openObjectStream}. If there is an IOException, it will throw it and the read will fail.
- *   </li>
+ * <li>
+ * If prefetch is enabled, PrefetchableTextFilesFirehose can fetch input objects in background.
+ * </li>
+ * <li> When next() is called, it first checks that there are already fetched files in local storage.
+ * <ol>
+ * <li>
+ * If exists, it simply chooses a fetched file and returns a {@link LineIterator} reading that file.
+ * </li>
+ * <li>
+ * If there is no fetched files in local storage but some objects are still remained to be read, the firehose
+ * fetches one of input objects in background immediately. If an IOException occurs while downloading the object,
+ * it retries up to the maximum retry count. Finally, the firehose returns a {@link LineIterator} only when the
+ * download operation is successfully finished.
+ * </li>
+ * </ol>
+ * </li>
+ * <li>
+ * If prefetch is disabled, the firehose returns a {@link LineIterator} which directly reads the stream opened by
+ * {@link #openObjectStream}. If there is an IOException, it will throw it and the read will fail.
+ * </li>
  * </ol>
  */
 public abstract class PrefetchableTextFilesFirehoseFactory<ObjectType>
```

Comment:
Would be nice to isolate fetching part of the logic from `FirehoseFactory` / line reading part of the logic, in separate classes.

Reply:
Good point. Added Fetcher for the fetching part.